core/analysis/text_token_stream.cpp

```diff
@@ -985,7 +985,8 @@ bool text_token_stream::reset(const string_ref& data) {
   if (state_->options.unicode == icu_locale_utils::Unicode::UTF8) {
     state_->data = icu::UnicodeString::fromUTF8(
       icu::StringPiece(data.c_str(), static_cast<int32_t>(data.size())));
-  } else if (state_->options.encoding == "utf-16") {
+  } else if (state_->options.encoding == "utf-16" || 
```

Comment:
Let's move encoding normalization to icu_locale_utils and check here only normalized values